thrift-interface-descriptions/component-cpis/user-profile-crud-cpi.thrift

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+/*
+ * Component Programming Interface definition for Apache Airavata User profile registry Service.
+ *
+*/
+
+include "../data-models/user-group-models/user_profile_model.thrift"
+include "registry_api_errors.thrift"
```

Comment:
Seems like the UserProfile service should have its own error types instead of using the registry error types. The way this is set up there are two RegistryServiceException classes that are generated.

I'm not sure what to recommend though. Maybe use the common airavata errors in airavata_errors.thrift? Or define a new UserProfileServiceException?